packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts

```diff
@@ -81,4 +81,31 @@ export abstract class ApexLibraryExecutor extends LibraryCommandletExecutor<{
       return result;
     };
   }
+
+  public getLogsWrapper(fn: (...args: any[]) => Promise<string[]>) {
+    const commandName = this.executionName;
+
+    return async function(...args: any[]): Promise<string[]> {
+      channelService.showCommandWithTimestamp(`Starting ${commandName}`);
+
+      const result = await vscode.window.withProgress(
+        {
+          title: commandName,
+          location: vscode.ProgressLocation.Notification
+        },
+        async () => {
+          // @ts-ignore
+          return (await fn.call(this, ...args)) as string[];
+        }
+      );
+
+      channelService.showCommandWithTimestamp(`Finished ${commandName}`);
+
+      const logPath = path.join(`${args[0].outputDir}`, `${args[0].logId}.log`);
```

Comment:
We're not returning the file path for the log that got retrieved as part of the library's response ?

Reply:
No I believe we're returning the logs themselves as the response

Reply:
Got it, we should take a look at the response and have it return everything we might need to consume so we don't have to do this on the consumer side. Let's log something for that in the library.